{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5b39b3cc-ee5b-4eb7-8918-b5e581cb357d",
   "metadata": {},
   "source": [
    "# lnc mRNA Filtering\n",
    "\n",
    "One of the main features of transcription is **alternative splicing (AS)**. This process allows one gene to produce differents mRNA products by modulating the AS machinery. Each unique alternative transcript is called an **isoform**, every gene can have one or more isoforms with different characteristics. Different isoforms can code for different proteins, commonly alternative isoforms have **premature stop codons (PTCs)** that funnel the transcripts to the degradation machinery (**NMD**). In this work we propose a **new function for mRNAs**, some isoforms never reach the ribosomes for translation and also are not degraded, making them an stable molecule that acummulates in the cell. We belive this isoforms have a role as lnc-mRNAs, mainly in the nucleus. This script is intended to filter out isoforms taking into account all the above mentiones characteristics, leaving us with a small group of lnc-mRNA isoforms for further studies.\n",
    "\n",
    "lnc RNAs will be filtered by two main features: Stable RNAs (not degraded) and not translated. Four datasets are used in this script:\n",
    "- Polyribosomes: https://doi.org/10.1111/tpj.12502\n",
    "- NMD deficient(Cicloheximide and upf1 & upf3 mutants): https://doi.org/10.1105/tpc.113.115485 \n",
    "- Own ONT data (Total, Nuclear, Citoplasmatic)\n",
    "- Own illumina data (Total, Nuclear, Citoplasmatic)\n",
    "\n",
    "**Objectives:**\n",
    "- Define a universe of isoforms (total isoforms present in the model studied)\n",
    "- Remove the isoforms that are being translated (polyribosomes)\n",
    "- Remove the isoforms that are being degraded (NMD deficient)\n",
    "    - To achieve this, isoforms that increase their presence on NMD deficient samples, are considered NMD sensitive\n",
    "- Keep the nuclear isoforms (and discard the citoplasmatic)\n",
    "- Asses light/Dark changes.\n",
    "- Quantify the proportion in nucleus/citoplasm of this isoforms\n",
    "\n",
    "*For this analisys I used a modified version of \"AtRTD2_QUASI\", called \"AtRTD2.1_QUASI\". This version was done by filtering out low expressed isoforms from 155 datasets quantified by SALMON using the \"AtRTD2_QUASI\" transcriptome (this datasets include illumina and ONT data).*\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5d80fe2b-a149-49d9-9c2b-4892e94e0084",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "from utils import concatenate_files_exclude"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "99049b29-430a-443f-8223-695c4c935694",
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'empias'",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mModuleNotFoundError\u001b[39m                       Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[3]\u001b[39m\u001b[32m, line 1\u001b[39m\n\u001b[32m----> \u001b[39m\u001b[32m1\u001b[39m \u001b[38;5;28;01mimport\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34;01mempias\u001b[39;00m\n",
      "\u001b[31mModuleNotFoundError\u001b[39m: No module named 'empias'"
     ]
    }
   ],
   "source": [
    "import empias"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1ac621a-0ee7-4d71-921f-80d869d90e95",
   "metadata": {},
   "source": [
    "## Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "58c262da-be77-4ba8-8c07-07c0688ae2ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_dir = \"/home/lucas-elytron/Documents/paper_doct/lnc_mrna/mapping/salmon/AtRTDv2_1_QUASI\"\n",
    "transcriptome = pd.read_parquet(\"/home/lucas-elytron/Documents/paper_doct/lnc_mrna/mapping/reference/mod_refs/AtRTDv2_1_QUASI.LS.parquet\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7f9fb6da-5da0-43e7-bbc2-d4e670a59f8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>TPM_CHX</th>\n",
       "      <th>TPM_MOCK</th>\n",
       "      <th>TPM_UPF1UPF3</th>\n",
       "      <th>TPM_UPF1</th>\n",
       "      <th>TPM_UPF3</th>\n",
       "      <th>TPM_WT</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th>isoform</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01010.1</th>\n",
       "      <td>3.626436</td>\n",
       "      <td>21.971265</td>\n",
       "      <td>24.489552</td>\n",
       "      <td>7.725926</td>\n",
       "      <td>8.610666</td>\n",
       "      <td>7.638263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">AT1G01020</th>\n",
       "      <th>AT1G01020_P1</th>\n",
       "      <td>10.880994</td>\n",
       "      <td>25.057372</td>\n",
       "      <td>16.052646</td>\n",
       "      <td>21.630980</td>\n",
       "      <td>16.291703</td>\n",
       "      <td>16.261399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01020_P2</th>\n",
       "      <td>0.348537</td>\n",
       "      <td>2.279401</td>\n",
       "      <td>0.358054</td>\n",
       "      <td>1.265387</td>\n",
       "      <td>0.601279</td>\n",
       "      <td>1.381167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01020_P3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.719321</td>\n",
       "      <td>2.606379</td>\n",
       "      <td>5.729312</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>8.076229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01020_P4</th>\n",
       "      <td>0.945659</td>\n",
       "      <td>1.320172</td>\n",
       "      <td>2.640832</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.046311</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATMG01330</th>\n",
       "      <th>ATMG01330.1</th>\n",
       "      <td>0.188437</td>\n",
       "      <td>0.203131</td>\n",
       "      <td>0.090114</td>\n",
       "      <td>0.357031</td>\n",
       "      <td>0.309601</td>\n",
       "      <td>0.101733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATMG01350</th>\n",
       "      <th>ATMG01350.1</th>\n",
       "      <td>0.198521</td>\n",
       "      <td>0.260449</td>\n",
       "      <td>0.661466</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.187362</td>\n",
       "      <td>0.205945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATMG01370</th>\n",
       "      <th>ATMG01370.1</th>\n",
       "      <td>5.622655</td>\n",
       "      <td>13.540405</td>\n",
       "      <td>10.619525</td>\n",
       "      <td>15.571162</td>\n",
       "      <td>6.142237</td>\n",
       "      <td>9.117486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATMG01380</th>\n",
       "      <th>ATMG01380.1</th>\n",
       "      <td>32.287186</td>\n",
       "      <td>72.567447</td>\n",
       "      <td>31.927099</td>\n",
       "      <td>117.233290</td>\n",
       "      <td>6.731542</td>\n",
       "      <td>48.946398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATMG01390</th>\n",
       "      <th>ATMG01390.1</th>\n",
       "      <td>108.767546</td>\n",
       "      <td>23.338974</td>\n",
       "      <td>58.644469</td>\n",
       "      <td>17.577448</td>\n",
       "      <td>11.798487</td>\n",
       "      <td>20.505417</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54038 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           TPM_CHX   TPM_MOCK  TPM_UPF1UPF3    TPM_UPF1  \\\n",
       "gene      isoform                                                         \n",
       "AT1G01010 AT1G01010.1     3.626436  21.971265     24.489552    7.725926   \n",
       "AT1G01020 AT1G01020_P1   10.880994  25.057372     16.052646   21.630980   \n",
       "          AT1G01020_P2    0.348537   2.279401      0.358054    1.265387   \n",
       "          AT1G01020_P3    0.000000   3.719321      2.606379    5.729312   \n",
       "          AT1G01020_P4    0.945659   1.320172      2.640832    0.000000   \n",
       "...                            ...        ...           ...         ...   \n",
       "ATMG01330 ATMG01330.1     0.188437   0.203131      0.090114    0.357031   \n",
       "ATMG01350 ATMG01350.1     0.198521   0.260449      0.661466    0.000000   \n",
       "ATMG01370 ATMG01370.1     5.622655  13.540405     10.619525   15.571162   \n",
       "ATMG01380 ATMG01380.1    32.287186  72.567447     31.927099  117.233290   \n",
       "ATMG01390 ATMG01390.1   108.767546  23.338974     58.644469   17.577448   \n",
       "\n",
       "                         TPM_UPF3     TPM_WT  \n",
       "gene      isoform                             \n",
       "AT1G01010 AT1G01010.1    8.610666   7.638263  \n",
       "AT1G01020 AT1G01020_P1  16.291703  16.261399  \n",
       "          AT1G01020_P2   0.601279   1.381167  \n",
       "          AT1G01020_P3   0.000000   8.076229  \n",
       "          AT1G01020_P4   3.046311   0.000000  \n",
       "...                           ...        ...  \n",
       "ATMG01330 ATMG01330.1    0.309601   0.101733  \n",
       "ATMG01350 ATMG01350.1    0.187362   0.205945  \n",
       "ATMG01370 ATMG01370.1    6.142237   9.117486  \n",
       "ATMG01380 ATMG01380.1    6.731542  48.946398  \n",
       "ATMG01390 ATMG01390.1   11.798487  20.505417  \n",
       "\n",
       "[54038 rows x 6 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "illumina_data, illumina_files = concatenate_files_exclude(file_dir, transcriptome, \"ONT_data\",)\n",
    "illumina_data = illumina_data.rename(columns={\"Name\":\"isoform\"})\n",
    "pivot_df = illumina_data.pivot_table(\n",
    "    index=['gene', 'isoform'],\n",
    "    columns='df_name', \n",
    "    values='TPM',\n",
    "    fill_value=float('nan')  # Use NaN for missing values\n",
    ")\n",
    "pivot_df.columns = ['TPM_' + col[4:-6] for col in pivot_df.columns]\n",
    "\n",
    "result_df = pivot_df\n",
    "result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5e0bf0e2-13d5-4d6e-92d8-10c9939a27e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>TPM_CHX</th>\n",
       "      <th>TPM_MOCK</th>\n",
       "      <th>TPM_UPF1UPF3</th>\n",
       "      <th>TPM_UPF1</th>\n",
       "      <th>TPM_UPF3</th>\n",
       "      <th>TPM_WT</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th>isoform</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">AT3G61860</th>\n",
       "      <th>AT3G61860_ID2</th>\n",
       "      <td>7.526846</td>\n",
       "      <td>21.839957</td>\n",
       "      <td>31.148346</td>\n",
       "      <td>35.591510</td>\n",
       "      <td>32.378725</td>\n",
       "      <td>23.847747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G61860_P1</th>\n",
       "      <td>3.061675</td>\n",
       "      <td>15.876204</td>\n",
       "      <td>11.322559</td>\n",
       "      <td>7.757980</td>\n",
       "      <td>6.432160</td>\n",
       "      <td>8.399215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G61860_s1</th>\n",
       "      <td>0.090624</td>\n",
       "      <td>1.398329</td>\n",
       "      <td>3.684376</td>\n",
       "      <td>5.838495</td>\n",
       "      <td>4.212439</td>\n",
       "      <td>6.825241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G61860_s2</th>\n",
       "      <td>12.375660</td>\n",
       "      <td>11.541558</td>\n",
       "      <td>47.315357</td>\n",
       "      <td>5.688878</td>\n",
       "      <td>4.989811</td>\n",
       "      <td>2.186952</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           TPM_CHX   TPM_MOCK  TPM_UPF1UPF3   TPM_UPF1  \\\n",
       "gene      isoform                                                        \n",
       "AT3G61860 AT3G61860_ID2   7.526846  21.839957     31.148346  35.591510   \n",
       "          AT3G61860_P1    3.061675  15.876204     11.322559   7.757980   \n",
       "          AT3G61860_s1    0.090624   1.398329      3.684376   5.838495   \n",
       "          AT3G61860_s2   12.375660  11.541558     47.315357   5.688878   \n",
       "\n",
       "                          TPM_UPF3     TPM_WT  \n",
       "gene      isoform                              \n",
       "AT3G61860 AT3G61860_ID2  32.378725  23.847747  \n",
       "          AT3G61860_P1    6.432160   8.399215  \n",
       "          AT3G61860_s1    4.212439   6.825241  \n",
       "          AT3G61860_s2    4.989811   2.186952  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_df.loc[[\"AT3G61860\",],:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0b16c907-5c12-4f3f-a9c0-be9d3caf05b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Nuc L</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Nuc D</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Cit L</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Cit D</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>TPM_1</th>\n",
       "      <th>TPM_2</th>\n",
       "      <th>TPM_3</th>\n",
       "      <th>TPM_1</th>\n",
       "      <th>TPM_2</th>\n",
       "      <th>TPM_3</th>\n",
       "      <th>TPM_1</th>\n",
       "      <th>TPM_2</th>\n",
       "      <th>TPM_3</th>\n",
       "      <th>TPM_1</th>\n",
       "      <th>TPM_2</th>\n",
       "      <th>TPM_3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th>isoform</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01010.1</th>\n",
       "      <td>2.759793</td>\n",
       "      <td>5.941394</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.101902</td>\n",
       "      <td>8.202830</td>\n",
       "      <td>6.580232</td>\n",
       "      <td>7.203409</td>\n",
       "      <td>7.886016</td>\n",
       "      <td>8.593391</td>\n",
       "      <td>5.974763</td>\n",
       "      <td>12.130097</td>\n",
       "      <td>10.128777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">AT1G01020</th>\n",
       "      <th>AT1G01020_P1</th>\n",
       "      <td>31.251582</td>\n",
       "      <td>55.162492</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>26.586747</td>\n",
       "      <td>38.920034</td>\n",
       "      <td>35.217447</td>\n",
       "      <td>31.214771</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>19.335129</td>\n",
       "      <td>9.957938</td>\n",
       "      <td>22.957437</td>\n",
       "      <td>10.128777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01020_P2</th>\n",
       "      <td>9.330350</td>\n",
       "      <td>16.049770</td>\n",
       "      <td>12.796301</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01020_P3</th>\n",
       "      <td>69.809793</td>\n",
       "      <td>47.615620</td>\n",
       "      <td>82.457146</td>\n",
       "      <td>78.852560</td>\n",
       "      <td>58.342096</td>\n",
       "      <td>92.000365</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>8.234241</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01020_P4</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "condition                   Nuc L                            Nuc D             \\\n",
       "sample                      TPM_1      TPM_2      TPM_3      TPM_1      TPM_2   \n",
       "gene      isoform                                                               \n",
       "AT1G01010 AT1G01010.1    2.759793   5.941394   0.000000   5.101902   8.202830   \n",
       "AT1G01020 AT1G01020_P1  31.251582  55.162492   0.000000  26.586747  38.920034   \n",
       "          AT1G01020_P2   9.330350  16.049770  12.796301   0.000000   0.000000   \n",
       "          AT1G01020_P3  69.809793  47.615620  82.457146  78.852560  58.342096   \n",
       "          AT1G01020_P4   0.000000   0.000000   0.000000   0.000000   0.000000   \n",
       "\n",
       "condition                              Cit L                          Cit D  \\\n",
       "sample                      TPM_3      TPM_1     TPM_2      TPM_3     TPM_1   \n",
       "gene      isoform                                                             \n",
       "AT1G01010 AT1G01010.1    6.580232   7.203409  7.886016   8.593391  5.974763   \n",
       "AT1G01020 AT1G01020_P1  35.217447  31.214771  0.000000  19.335129  9.957938   \n",
       "          AT1G01020_P2   0.000000   0.000000  0.000000   0.000000  0.000000   \n",
       "          AT1G01020_P3  92.000365   0.000000  0.000000   0.000000  0.000000   \n",
       "          AT1G01020_P4   0.000000   0.000000  0.000000   0.000000  0.000000   \n",
       "\n",
       "condition                                     \n",
       "sample                      TPM_2      TPM_3  \n",
       "gene      isoform                             \n",
       "AT1G01010 AT1G01010.1   12.130097  10.128777  \n",
       "AT1G01020 AT1G01020_P1  22.957437  10.128777  \n",
       "          AT1G01020_P2   0.000000   0.000000  \n",
       "          AT1G01020_P3   8.234241   0.000000  \n",
       "          AT1G01020_P4   0.000000   0.000000  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ONT_data, ONT_files = concatenate_files_exclude(file_dir, transcriptome, \"PRJNA176940\")\n",
    "\n",
    "grouped = ONT_data.groupby('df_name')\n",
    "ont_dfs, ont_dfs_names  = {x : grouped.get_group(x) for x in grouped.groups}, [x for x in grouped.groups]\n",
    "ont_barcode_keys = [(\"Nuc L\",\"TPM_1\"),(\"Nuc L\",\"TPM_2\"),(\"Nuc L\",\"TPM_3\"),\n",
    "                    (\"Nuc D\",\"TPM_1\"),(\"Nuc D\",\"TPM_2\"),(\"Nuc D\",\"TPM_3\"),\n",
    "                    (\"Cit L\",\"TPM_1\"),(\"Cit L\",\"TPM_2\"),(\"Cit L\",\"TPM_3\"),\n",
    "                    (\"Cit D\",\"TPM_1\"),(\"Cit D\",\"TPM_2\"),(\"Cit D\",\"TPM_3\"),]\n",
    "ONT_nuc_cito = pd.concat([ont_dfs[x].TPM for x in ont_dfs_names], axis=1, keys=ont_barcode_keys)\n",
    "#ONT_total = pd.concat([ont_dfs[x].TPM for x in ont_dfs_names[-2:]], axis=1, keys=[(\"Total D\",\"TPM_1\"),(\"Total L\",\"TPM_1\")])\n",
    "#Check\n",
    "#ONT_total.columns.set_names([\"condition\", \"sample\"], inplace=True)\n",
    "ONT_nuc_cito.columns.set_names([\"condition\", \"sample\"], inplace=True)\n",
    "ONT_nuc_cito.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "958d1936-5144-4067-b8f0-1110edbe7021",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Nuc L</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Nuc D</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Cit L</th>\n",
       "      <th colspan=\"3\" halign=\"left\">Cit D</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>TPM_1</th>\n",
       "      <th>TPM_2</th>\n",
       "      <th>TPM_3</th>\n",
       "      <th>TPM_1</th>\n",
       "      <th>TPM_2</th>\n",
       "      <th>TPM_3</th>\n",
       "      <th>TPM_1</th>\n",
       "      <th>TPM_2</th>\n",
       "      <th>TPM_3</th>\n",
       "      <th>TPM_1</th>\n",
       "      <th>TPM_2</th>\n",
       "      <th>TPM_3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th>isoform</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">AT3G61860</th>\n",
       "      <th>AT3G61860_ID2</th>\n",
       "      <td>136.466411</td>\n",
       "      <td>135.012686</td>\n",
       "      <td>311.250844</td>\n",
       "      <td>557.108229</td>\n",
       "      <td>550.914695</td>\n",
       "      <td>558.654675</td>\n",
       "      <td>2.707569</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>12.336501</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>11.070926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G61860_P1</th>\n",
       "      <td>43.749715</td>\n",
       "      <td>52.408766</td>\n",
       "      <td>53.185616</td>\n",
       "      <td>43.947466</td>\n",
       "      <td>31.514748</td>\n",
       "      <td>51.051770</td>\n",
       "      <td>74.128790</td>\n",
       "      <td>104.001051</td>\n",
       "      <td>84.234626</td>\n",
       "      <td>41.436362</td>\n",
       "      <td>46.787517</td>\n",
       "      <td>59.830515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G61860_s1</th>\n",
       "      <td>19.868876</td>\n",
       "      <td>26.468734</td>\n",
       "      <td>9.250141</td>\n",
       "      <td>51.987760</td>\n",
       "      <td>61.765164</td>\n",
       "      <td>52.703543</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.460159</td>\n",
       "      <td>5.995978</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G61860_s2</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.485313</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "condition                     Nuc L                               Nuc D  \\\n",
       "sample                        TPM_1       TPM_2       TPM_3       TPM_1   \n",
       "gene      isoform                                                         \n",
       "AT3G61860 AT3G61860_ID2  136.466411  135.012686  311.250844  557.108229   \n",
       "          AT3G61860_P1    43.749715   52.408766   53.185616   43.947466   \n",
       "          AT3G61860_s1    19.868876   26.468734    9.250141   51.987760   \n",
       "          AT3G61860_s2     0.000000    0.000000    0.000000    0.000000   \n",
       "\n",
       "condition                                            Cit L              \\\n",
       "sample                        TPM_2       TPM_3      TPM_1       TPM_2   \n",
       "gene      isoform                                                        \n",
       "AT3G61860 AT3G61860_ID2  550.914695  558.654675   2.707569    0.000000   \n",
       "          AT3G61860_P1    31.514748   51.051770  74.128790  104.001051   \n",
       "          AT3G61860_s1    61.765164   52.703543   0.000000    2.460159   \n",
       "          AT3G61860_s2     1.485313    0.000000   0.000000    0.000000   \n",
       "\n",
       "condition                               Cit D                        \n",
       "sample                       TPM_3      TPM_1      TPM_2      TPM_3  \n",
       "gene      isoform                                                    \n",
       "AT3G61860 AT3G61860_ID2   0.000000  12.336501   0.000000  11.070926  \n",
       "          AT3G61860_P1   84.234626  41.436362  46.787517  59.830515  \n",
       "          AT3G61860_s1    5.995978   0.000000   0.000000   0.000000  \n",
       "          AT3G61860_s2    0.000000   0.000000   0.000000   0.000000  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ONT_nuc_cito.loc[[\"AT3G61860\",],:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f12d4a9d-4403-4a2a-8fe1-cbd0b16e3b79",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81b82330-6ebf-4acc-b7de-527bcda6932a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "empias",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
